test_util/fault_injection_test_fs.h

```diff
@@ -238,14 +278,91 @@ class FaultInjectionTestFS : public FileSystemWrapper {
     error_ = io_error;
   }
 
+  // Specify what the operation, so we can inject the right type of error
+  enum ErrorOperation {
+    READ = 0,
+    OPEN,
+  };
+
+  // Set thread-local parameters for error injection. The first argument,
+  // seed is the seed for the random number generator, and one_in determines
+  // the probability of injecting error (i.e an error is injected with
+  // 1/one_in probability)
+  void SetThreadLocalReadError(uint32_t seed, int one_in) {
+    struct ErrorContext* ctx =
```

Comment:
Any particular reason for C-style `struct ErrorContext` instead of C++ style `ErrorContext`?